patches/net/minecraft/server/MinecraftServer.java.patch

```diff
@@ -292,7 +292,12 @@
 +                else 
 +                {
 +                    worldType = env.toString().toLowerCase();
-+                    name = "DIM" + dimension;
++                    // Handle forge worlds
++                    name = provider.getSaveFolder();
++                    // Handle plugin worlds
++                    if (name == null || name.equals("DIM0") || name.equals("DIM-1") || name.equals("DIM1")) {
```

Comment:
It can be only those 3 names? Where do they come from?

Reply:
If the worldprovider is one of the vanilla ones reused by a plugin for a custom world and does not define its own save folder, those are the names.